### DISCUSSION TITLE
Snap Prioritization

### DISCUSSION THREAD

**Discussion by Ichisich** [RamonUnch/AltDrag#499] [Ideas]
Currently, AltSnap only considers the edges of the last focused window before the current window when it comes to snapping.

e.g.
![1](https://github.com/RamonUnch/AltSnap/assets/31080003/aede5015-9e6a-4344-9c59-72aee72ee1f5)

When window 2 was focused before snapping, window 1 snaps like this into the top left corner:
![2l](https://github.com/RamonUnch/AltSnap/assets/31080003/a76810ab-82f0-4462-90c3-c0c804dbb2b5)
top right corner:
![2r](https://github.com/RamonUnch/AltSnap/assets/31080003/381c2afe-4fa7-4f38-aae4-6f37885c685e)

When window 3 was focused, snapping window 1 into the top left corner looks like this:
![3l](https://github.com/RamonUnch/AltSnap/assets/31080003/4faada7c-57dc-4575-9f0b-14505fdce21d)
top right corner:
![3r](https://github.com/RamonUnch/AltSnap/assets/31080003/1d74553c-6a9a-401d-bd2d-1ba5386c69f0)

Personally, I would prefer if image 2 and 5 would always be the outcome, but I can imagine that it is highly subjective.
A cycling approach could also be applied, in which snapping cycles through the order of the prior focused windows and snaps accordingly.

**Comment by RamonUnch** [comment]
The reason it considers the last focused window first is because if you had several layers of snapped windows it would become exponentially more complicated to choose where to put the window.

However what you present should be fixable doable without too much trouble.

**Reply by Ichisich** [reply]
In my favored scenario, AltSnap would also only have to consider one window but prioritize the window which lies horizontally opposed to the snap direction, but not vertically.

![Snap](https://github.com/RamonUnch/AltSnap/assets/31080003/64bed39e-498f-4b7a-b5c2-323ca808310b)

When snapping a theoretical fifth window into the: 
- top left, consider the TL window
- bottom left, consider BL window
- top right, consider the TR window
- bottom right, considert the BR window
